src/main/java/com/appdynamics/extensions/webspheremq/metricscollector/InquireQCmdCollector.java

```diff
@@ -72,9 +72,15 @@ protected void publishMetrics() throws TaskExecutionException {
                 processPCFRequestAndPublishQMetrics(queueGenericName, request,COMMAND);
             } catch (PCFException pcfe) {
                 logger.error("PCFException caught while collecting metric for Queue: {} for command {}",queueGenericName,COMMAND, pcfe);
-                PCFMessage[] msgs = (PCFMessage[]) pcfe.exceptionSource;
-                for (int i = 0; i < msgs.length; i++) {
-                    logger.error(msgs[i].toString());
+                if (pcfe.exceptionSource instanceof PCFMessage[]) {
```

Comment:
I was surprised to see having to cast a field within an exception;  after all, the PCF has been decades old.   

While testing, I did just run into ClassCastException.   So I guess PR approved!